pkg/dockerregistry/server/simplelookupicsp_test.go

```diff
@@ -337,29 +284,56 @@ func TestFirstRequest(t *testing.T) {
 					Name:      "image",
 				},
 			},
-			rules: []runtime.Object{
-				&v1alpha1.ImageContentSourcePolicy{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "icsp-rule",
-					},
-					Spec: v1alpha1.ImageContentSourcePolicySpec{
-						RepositoryDigestMirrors: []v1alpha1.RepositoryDigestMirrors{
-							{
-								Source: "i.do.not.exist/repo/image",
-								Mirrors: []string{
-									"-92</asfg",
-								},
-							},
-						},
+			rules: []rule{
+				{
+					name: "rule",
+					ruleElement: []element{
+						{source: "i.do.not.exist/repo/image",
+							mirrors: []string{
+								"-92</asfg",
+							}},
 					},
 				},
 			},
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
-			cli := fake.NewSimpleClientset(tt.rules...)
+			icspRules := []runtime.Object{}
+			for _, rule := range tt.rules {
+				icspRules = append(icspRules, newICSPRule(rule))
+			}
+			cli := fake.NewSimpleClientset(icspRules...)
+			cfgcli := cfgfake.NewSimpleClientset()
+			lookup := NewSimpleLookupICSPStrategy(
+				cli.OperatorV1alpha1().ImageContentSourcePolicies(),
+				cfgcli.ConfigV1().ImageDigestMirrorSets(),
+			)
+
+			ref, err := reference.Parse(tt.ref)
+			if err != nil {
+				t.Fatalf("unexpected error parsing reference: %s", err)
+			}
+
+			alternates, err := lookup.FirstRequest(context.Background(), ref)
+			if err != nil {
+				t.Fatalf("FirstRequest does not return error, received: %s", err)
+			}
+
+			if !reflect.DeepEqual(alternates, tt.res) {
+				t.Errorf("expected %+v, received %+v", tt.res, alternates)
+			}
+		})
+
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Can you append something to the test case name to indicate that this sub-test is for IDSM rules?
Same goes for the ICSP rules sub-test above on line 300.